common/modules/gcp-project/main.tf

```diff
@@ -25,120 +25,48 @@ variable "serviceaccount_key" {}
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/compute.admin"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/container.clusterAdmin"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/container.admin"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/dns.admin"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-      "serviceAccount:projectowner@${var.project_id}.iam.gserviceaccount.com",
-    ]
-  }
-
-  binding {
-    role = "roles/iam.serviceAccountKeyAdmin"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/iam.serviceAccountUser"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/logging.configWriter"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/monitoring.editor"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/resourcemanager.projectIamAdmin"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
-
-  binding {
-    role = "roles/serviceusage.serviceUsageAdmin"
-
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-    ]
-  }
+variable "projectowner_roles" {
+  type = "list"
+
+  default = [
+    "roles/cloudkms.admin",
+    "roles/cloudkms.cryptoKeyEncrypterDecrypter",
+    "roles/compute.admin",
+    "roles/container.clusterAdmin",
+    "roles/container.admin",
+    "roles/dns.admin",
+    "roles/iam.serviceAccountKeyAdmin",
+    "roles/iam.serviceAccountUser",
+    "roles/logging.configWriter",
+    "roles/monitoring.editor",
+    "roles/resourcemanager.projectIamAdmin",
+    "roles/serviceusage.serviceUsageAdmin",
+    "roles/storage.admin",
+    "roles/owner",
+  ]
+}
 
-  binding {
-    role = "roles/storage.admin"
+variable "common_projectowner_roles" {
+  type = "list"
 
-    members = [
-      "serviceAccount:${google_service_account.project.email}",
-      "serviceAccount:projectowner@${var.project_id}.iam.gserviceaccount.com",
-    ]
-  }
+  default = [
+    "roles/dns.admin",
+    "roles/storage.admin",
+  ]
+}
 
-  binding {
-    role = "roles/owner"
+resource "google_project_iam_member" "common_projectowner" {
+  count   = "${length(var.common_projectowner_roles)}"
+  project = "${google_project.project.project_id}"
+  role    = "${var.common_projectowner_roles[count.index]}"
+  member  = "serviceAccount:projectowner@${var.project_id}.iam.gserviceaccount.com"
+}
```

Comment:
I would strongly prefer to use `google_iam_policy` over `google_project_iam_member`. The difference is that the 1st one is authoritative (ie. it will remove unwanted additional permissions), while the second one is not.

Reply:
We can not use authoritative policy setting here, because it would override roles for SAs that we're creating on the fly. But I agree current implementation is not ideal.